### PR TITLE
Create target-platforms/boards page with a new template type, adapt existing content

### PR DESCRIPTION
**Pull request by caldav**
This PR introduces the "Boards" page. The goal of this page is to provide a searchable list of development boards that supports Ubuntu, regardless of the Ubuntu version (core/desktop/server) they are running. Results are clickable, linking to install steps and image downloads for each board.

## Testing
    ./run
http://0.0.0.0:8015/target-platforms/boards

## Changes
* The existing boards related pages have been adapted with new metadata and anchors.
* This introduces a new "grid_boards" component that provides a search/filter functionality on the grid
* The "summary_refine" page type is based on the "summary" type and provides "grid_boards"
* The "target-platforms" component has been renamed to "target-platforms-featured" to be more accurate

gif

## Default view
screenshot from 2017-09-19 17-29-46

## Tag search
screenshot from 2017-09-19 17-30-13

## Title search
screenshot from 2017-09-19 17-30-01

### CODE REVIEW

File: templates/pages/target-platforms/boards/index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - /core/get-started/artik-5-10
+    - /core/get-started/dragonboard-410c
+    - /core/get-started/intel-nuc
+    - /core/get-started/intel-joule
+    - /core/get-started/kvm
+----
+
+# Boards
+
+All these boards are supported with official Ubuntu images, but they didn't necessarily went through the Ubuntu Certification program. You can find Ubuntu Certified hardware at [certification.ubuntu.com](https://certification.ubuntu.com).
```
Comment (evilnick) [inline templates/pages/target-platforms/boards/index.md]: I would say:"
All of these boards are supported with official Ubuntu images, but they may not have completed the Ubuntu Certification program. You can find Ubuntu Certified hardware at [certification.ubuntu.com (https://certification.ubuntu.com).

Comment (caldav) [reply templates/pages/target-platforms/boards/index.md]: +1

**Review by evilnick — APPROVED** [review]
+1 from me

**Comment by pmahnke** [comment]
@caldav love the animated gif!  how did you do that?

**Comment by caldav** [comment]
@pmahnke https://github.com/phw/peek (the dev also published a "peek" snap in the edge channel, I haven't tested it yet)

**Comment by caldav** [comment]
I forgot to say in the description: this work is a first step to accommodate the arrival of at least two new boards/images in the coming weeks, that won't be running core, but that we still have to highlight.

File: templates/includes/components/grid_boards.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {% for page in pages %}
+        <li class="col-6 grid-list__item{% if forloop.last %} last-item{% endif %}">
+          <h3>{{ page.title }}</h3>
+          <!-- Do not display tags on grid items -->
+          <p class="board-tags" style="display:none;">{{ page.tags }}</p>
+          <div class="grid-list__description">
+            <ul class="p-list--divided">
+              {% for d in page.distributions %}
+                <li class="p-list__item  is-ticked">
+                    <!--
```
Comment (WillMoggridge) [inline templates/includes/components/grid_boards.html]: Can these comments be changed to a Django comment block, they currently get printed out on the page many times.
```
{% comment %}
comments here
{% endcomment %}
```

**Review by WillMoggridge — CHANGES_REQUESTED** [review]
This is pretty awesome!

Just the one point for code, a request to change the comments a bit.
We also should get design to give it a little look.

File: templates/includes/components/grid_boards.html (hunk 2 above)
Comment (caldav) [reply templates/includes/components/grid_boards.html]: Hah good point, I forgot about the "comment" tag. Changing.

**Comment by WillMoggridge** [comment]
The failed check seems to be from a cert error on https://rocket.ubuntu.com, I am going to raise the issue now and it should not block this.

(Still want to get a designer to check)

**Review by WillMoggridge — APPROVED** [review]

**Comment by spencerbygraves** [comment]
Thanks @caldav please can you use the close icon from the icon set:

https://vanilla-framework.github.io/vanilla-framework/examples/patterns/icons/icons-light/

**Comment by caldav** [comment]
@spencerbygraves wfy? 
screenshot from 2017-09-20 18-00-49

**Comment by spencerbygraves** [comment]
@caldav looks good, thanks for updating it.

**Pull request merged**